Makefile.kokkos-kernels

```diff
@@ -306,6 +309,16 @@ ifeq (${KOKKOSKERNELS_INTERNAL_ENABLE_CUBLAS}, 1)
   tmp := $(shell echo "\#define KOKKOSKERNELS_ENABLE_TPL_CUBLAS" >> KokkosKernels_config.tmp )
 endif
 
+ifeq (${KOKKOSKERNELS_INTERNAL_ENABLE_MAGMA}, 1)
+  KOKKOS_INTERNAL_NVCC_PATH := $(shell which nvcc)
+  CUDA_PATH ?= $(KOKKOS_INTERNAL_NVCC_PATH:/bin/nvcc=)
+  KOKKOSKERNELS_CPPFLAGS += -DADD_ -I${MAGMA_PATH}/include -I${MAGMA_PATH}/sparse/include -I${CUDA_PATH}/include
```

Comment:
-DADD_ means that the fortran linker uses a single underbar in the magma. This could be different based on the compiler. You should not add this flag here and allow users to provide how the fortran is going to be linked.

Reply:
@kyungjoo-kim Thanks. I have made a change so that users can provide (through KOKKOSKERNELS_OPTIONS) how Fortran functions are name-mangled using the available logic for Fortran mangling macros.

How can I add this cxx flag in the cmake file?

Reply:
In the makefile, there is no way to detect the fortran mangling interface. Check the Makefile.kokkos-kernels. There are
```
KOKKOSKERNELS_INTERNAL_OVERRIDE_FORTRAN_MANGLING_WITH_BLAHBLAH_UNDERSCORE
```
Define -DADD flag based on the macro definition that users provide. 

In CMake, there is a similar mechanism testing FC_FN_UNDERSCORE. See cmake/tribits/core/config_tests/FortranMangling.cmake. The value is supposed to be one of NO_UNDER, UNDER, SECOND_UNDER. Testing the value and add an appropriate compile flag.

Reply:
@kyungjoo-kim I already updated the Makefile.kokkos-kernels using ```KOKKOSKERNELS_INTERNAL_OVERRIDE_FORTRAN_MANGLING_WITH_BLAHBLAH_UNDERSCORE```. Please see the latest version.

I will fix CMake too.

Reply:
@kyungjoo-kim I have realized that this fix for CMake is in the Trilinos repo, which I am not sure I can create a pull request. Can you help me with fixing and testing in Trilinos?

Reply:
@kyungjoo-kim I have added the cxx flag to CMakeLists.txt and tested this PR in Trilinos with MAGMA TPL enabling. Please see my comment below.